compiler/src/iree/compiler/Codegen/Common/test/block_dynamic_dims.mlir

```diff
@@ -0,0 +1,101 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-codegen-block-dynamic-dimensions, cse))" --split-input-file --mlir-print-local-scope %s | FileCheck %s
+
+#pipeline_layout = #hal.pipeline.layout<constants = 4, bindings = [
+    #hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">,
+    #hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">,
+    #hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">,
+    #hal.pipeline.binding<storage_buffer, "ReadOnly|Indirect">,
+    #hal.pipeline.binding<storage_buffer, Indirect>], flags = Indirect>
+func.func @block_attention_dims() {
+  %c0 = arith.constant 0 : index
+  %cst = arith.constant 8.837890e-02 : f16
+  %m_in = hal.interface.constant.load layout(#pipeline_layout) ordinal(0) : index
+  %k2_in = hal.interface.constant.load layout(#pipeline_layout) ordinal(1) : index
+  %0:2 = util.assume.int
+      %m_in<umin = 16, umax = 4080, udiv = 16>,
+      %k2_in<umin = 16, umax = 4080, udiv = 32>
+    : index, index
+  %m = flow.dispatch.workload.ordinal %0#0, 0 : index
+  %k2 = flow.dispatch.workload.ordinal %0#1, 1 : index
+  %q_in = hal.interface.binding.subspan layout(#pipeline_layout) binding(0) alignment(64) offset(%c0) flags("ReadOnly|Indirect")
+      : !flow.dispatch.tensor<readonly:tensor<4x?x32x128xf16>>{%m}
+  %key_in = hal.interface.binding.subspan layout(#pipeline_layout) binding(1) alignment(64) offset(%c0) flags("ReadOnly|Indirect")
+      : !flow.dispatch.tensor<readonly:tensor<4x?x32x128xf16>>{%k2}
+  %value_in = hal.interface.binding.subspan layout(#pipeline_layout) binding(2) alignment(64) offset(%c0) flags("ReadOnly|Indirect")
+      : !flow.dispatch.tensor<readonly:tensor<4x?x32x128xf16>>{%k2}
+  %mask_in = hal.interface.binding.subspan layout(#pipeline_layout) binding(3) alignment(64) offset(%c0) flags("ReadOnly|Indirect")
+      : !flow.dispatch.tensor<readonly:tensor<4x32x?x?xf16>>{%m, %k2}
+  %output_in = hal.interface.binding.subspan layout(#pipeline_layout) binding(4) alignment(64) offset(%c0) flags(Indirect)
+      : !flow.dispatch.tensor<writeonly:tensor<4x?x32x128xf16>>{%m}
```

Comment:
nit: The dispatch bindings could be moved to function operands

Reply:
Yeah, but want to also check that these get folded.

Reply:
Ah right, I forgot these are updated too